packages/flutter/lib/src/widgets/restoration.dart

```diff
@@ -0,0 +1,1031 @@
+///
+/// In addition to being invoked right after [State.initState], [restoreState]
+/// is invoked again when new restoration data has been provided to the mixin.
+/// When this happens, the [State] object must re-register all properties with
+/// [registerForRestoration] again to restore them to their previous values as
+/// described by the new restoration data. All initialization logic that depends
+/// on the current value of a restorable property should be included in the
+/// [restoreState] method to ensure it re-executes when the properties are
+/// restored to a different value during the life time of the [State] object.
+///
+/// Internally, the mixin stores the restoration data from all registered
+/// properties in a [RestorationBucket] claimed from the surrounding
+/// [RestorationScope] using the [State]-provided [restorationId]. The
+/// [restorationId] must be unique in the surrounding [RestorationScope]. State
+/// restoration is disabled for the [State] object using this mixin if
+/// [restorationId] is null or when there is no surrounding [RestorationScope].
+/// In that case, the values of the registered properties will not be restored
+/// during state restoration.
+///
+/// The [RestorationBucket] used to store the registered properties is available
+/// via the [bucket] getter. Interacting directly with the bucket is uncommon,
+/// but the [State] object may make this bucket available for its descendants to
+/// claim child buckets from. For that, the [bucket] is injected into the widget
+/// tree in [State.build] with the help of a [UnmanagedRestorationScope].
+///
+/// The [bucket] getter returns null if state restoration is turned off. If
+/// state restoration is turned on or off during the lifetime of the widget
+/// (e.g. because [restorationId] changes from null to non-null) the value
+/// returned by the getter will also change from null to non-null or vice versa.
+/// The mixin calls [didToggleBucket] on itself to notify the [State] object
+/// about this change. Overriding this method is not necessary as long as the
+/// [State] object does not directly interact with the [bucket].
+///
+/// Whenever the value returned by [restorationId] changes,
+/// [didUpdateRestorationId] must be called (unless the change already triggers
+/// a call to [didUpdateWidget]).
+///
+/// {@tool dartpad --template=freeform}
+/// This example demonstrates how to make a simple counter app restorable by
+/// using the [RestorationMixin] and a [RestorableInt].
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart main
+/// void main() => runApp(RestorationExampleApp());
+/// ```
+///
+/// ```dart preamble
+/// class RestorationExampleApp extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     // TODO(goderbauer): remove the [RootRestorationScope] once it is part of [MaterialApp]. // ignore: flutter_style_todos
```

Comment:
maybe make this a comment rather than part of the template (i.e. remove the `///` on that line)

Reply:
Turns out, the dartdoc parser doesn't like that.

Reply:
This TODO is causing an error on triple-head (tip-of-tree Dart, engine, framework) flutter analyze tests at https://ci.chromium.org/p/dart/builders/ci.sandbox/flutter-engine-linux/11403

Reply:
What's the failure? In the logs from the linked run I am only seeing directory-related failures (e.g. `No such file or directory`). How are those caused by a TODO in a comment?

Reply:
The "flutter analyze" step with log of stdout at https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8873492683668712848/+/steps/flutter_analyze/0/stdout
has a failure due to that TODO

Reply:
I am removing the TODO here: https://github.com/flutter/flutter/pull/62500